packages/api/app/Models/User.js

```diff
@@ -38,17 +38,50 @@ class User extends Model {
 			role,
 			status,
 			company,
+			zipcode,
+			cpf,
+			birth_date,
+			phone_number,
+			lattes_id,
+			address,
+			address2,
+			district,
+			city,
+			state,
+			country,
 		} = payload;
 
 		const data = {
 			first_name,
 			last_name,
 			email,
 			password,
+			zipcode,
+			cpf,
+			birth_date,
+			phone_number,
+			lattes_id,
+			address,
+			address2,
+			district,
+			city,
+			state,
+			country,
 		};
 
 		if (status) data.status = status;
 		if (company) data.company = company;
+		if (zipcode) data.zipcode = zipcode;
```

Comment:
@alexandreadames , minha dúvida que falei era essa aqui: você verifica (por exemplo), se `zipcode` existe e, se existir, insere ele no `data`. Mas ele não já foi inserido no data mais acima, na linha 59?

Reply:
Está certo Luiz, o data deve ser assim:
```javascript
const data = {
			first_name,
			last_name,
			email,
			password,
		};
```
Só para você entender melhor o payload é tudo que eu mando para `User.create(payload)`, inclusive role, que não necessariamente um campo do model User. @nicholasio havia sugerido eu fazer `modelInstance.fill(payload)` mas nesse caso teríamos um problema com campos como status que teriam um valor padrão no banco, através dos ifs eu garanto que apenas campos pasados seram salvos no banco. Eu não tenho como prever quais campos seram passados no create por isso eu faço tratamento dos campos opcionais através dos ifs.

Reply:
Entendi. Mas acho que poderíamos ter algo (se é que você não testou). Se tem um campo que é opcional, por exemplo, `zip_code` e ele não for enviado, **acho** (precisa verificar) que ele vai chegar lá no `payload` como `undefined`, certo? Então, quando ele for enviado pro banco, é como se ele não existisse, não? Se isso for verdade, você pode simplemente fazer o `modelInstance.fill(payload)`. 

Em relação ao `role`, ele não está sendo salvo somente depois de os dados da própria tabela de user terem sido salvos?
```
       modelInstance.fill(data);
        await modelInstance.save();
		const userRole = await Role.getRole(role || roles.DEFAULT_USER);
		await modelInstance.role().associate(userRole);
		return modelInstance;
```

Reply:
Eu tinha testado isso antes e tive um problema com status. Ex.

```javascript
const adminUser = {
	email: 'sabiaadminuser@gmail.com',
	password: '123123',
	first_name: 'FirstName',
	last_name: 'LastName',
	role: 'ADMIN',
};
```
Se eu der um `User.create(adminUser)` dá forma como você disse ele irá criar um usuário admin com status null e não pending. Outra coisa o fill ele preenche com campos que existem na tabela do usuário, o role não é um campo do model user portanto não pode ser salvo diretamente.

Reply:
Hum...verdade. Em relação ao `role`, não tinha atentado pra isso. Em relação ao `status`, é uma situação específica? Digo, é porque ele tem valores pré-definidos? Se for, uma melhor solução seria
```
const {status, role, ...userData} = payload;
```
Ou seja, você tirava da jogada somente os campos que precisam de tratamento diferenciado. `userData` vai ter o restante dos campos. Daí, você tratava somente eles:
```
if (status) userData.status = status;

[...]

modelInstance.fill(userData);
await modelInstance.save();
const userRole = await Role.getRole(role || roles.DEFAULT_USER);
await modelInstance.role().associate(userRole);
return modelInstance;
```

Ou seja, ao invés de tratar todos os campos, você tratava somente aqueles que precisam de tratamento especial. O que acha?

Reply:
Ótima ideia, vou testar aqui.

Reply:
Ficou bem melhor :+1: